spec/lib/industrialist/manufacturable_spec.rb

```diff
@@ -1,28 +1,46 @@
 require 'spec_helper'
 
 RSpec.describe Industrialist::Manufacturable do
-  before do
+  def undefine_constants(*consts)
+    consts.each { |const| Object.send(:remove_const, const) }
+  end
+
+  around do |example|
```

Comment:
I think this is actually a perfect use case for `before(:all)` since we only want these consts defined once in each describe block:
```ruby
RSpec.describe Industrialist::Manufacturable do
  before(:all) do
    class Automobile
      include Industrialist::Manufacturable
      create_factory :AutomobileFactory
    end
    class Book
      include Industrialist::Manufacturable
      create_factory :BookFactory
    end
  end

  describe '.create_factory' do
    it 'assigns the provided name to the factory instance' do
      expect(AutomobileFactory).to be_an(Industrialist::Factory)
    end
  end

  describe '.corresponds_to' do
    before(:all) do
      class Sedan < Automobile
        corresponds_to :sedan
      end
      class Paperback < Book
        corresponds_to :paperback
      end
    end

    it 'registers the class under the provided key' do
      expect(AutomobileFactory.registry[:sedan]).to equal(Sedan)
      expect(BookFactory.registry[:paperback]).to equal(Paperback)
    end

    it 'registers the class only under the provided key' do
      expect(AutomobileFactory.registry[:paperback]).to be_nil
      expect(BookFactory.registry[:sedan]).to be_nil
    end
  end
end
```

That way we don't have to worry about removing const after each block. Rspec will take care of it tearing them down.